File: MergeTwoSortedLists/mtsl.js


function ListNode(val) {
  this.val = val;
  this.next = null;
}

/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
  var result = [];
  while(l1 !== null || l2 !== null) {

    if(!l1) {
      result.push(l2.val);
      l2 = l2.next;
      continue;
    } else if(!l2) {
      result.push(l1.val);
      l1 = l1.next;
      continue;
    }
    if(l1.val <= l2.val) {
      result.push(l1.val);
      l1 = l1.next;
    } else {
      result.push(l2.val);
      l2 = l2.next;
    }
  }
  return result;
};

File: MergeTwoSortedLists/mtsl.java
public class ListNode {
   int val;
   ListNode next;
   ListNode(int x) { val = x; }
}
public class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
      List result = new ArrayList<Integer>();
      while(l1 != null || l2 != null) {
        if(l1 == null) {
          result.add(l2.val);
          l2 = l2.next;
        } else if(l2 == null) {
          result.add(l1.val);
          l1 = l1.next;
        }

        if(l1.val <= l2.val) {
          result.add(l1.val);
          l1= l1.next;
        } else {
          result.add(l2.val);
          l2 = l2.next;
        }
      }
      return result;
    }
}
